crates/primitives-traits/src/transaction/signed.rs

```diff
@@ -77,3 +82,23 @@ pub trait SignedTransaction:
     /// Fills [`TxEnv`] with an [`Address`] and transaction.
     fn fill_tx_env(&self, tx_env: &mut TxEnv, sender: Address);
 }
+
+impl<T: SignedTransaction> AlloyTransactionExt for T {
+    type Type = <T::Transaction as AlloyTransactionExt>::Type;
+
+    fn signature_hash(&self) -> B256 {
+        self.transaction().signature_hash()
+    }
+
+    fn is_dynamic_fee(&self) -> bool {
+        self.transaction().is_dynamic_fee()
+    }
```

Comment:
imo this can be moved into `alloy_consensus::Transaction`, wdyt?

Reply:
yep, this should be moved

Reply:
https://github.com/alloy-rs/alloy/pull/1638